Trappist/src/Promact.Trappist.Repository/Test/ITestsRepository.cs

```diff
@@ -10,15 +10,15 @@ public interface ITestsRepository
         /// this method is used to create a new test
         /// </summary>
         /// <param name="test">object of Test</param>
-        void CreateTest(Test test);
+      Task CreateTestAsync(Test test);
 
         /// <summary>
         /// this method is used to verify Name of the test is unique or not
         /// </summary>
-        /// <param name="test">object of Test</param>
+        /// <param name="testName">name of the test</param>
         /// <returns>boolean</returns>
-        bool UniqueTestName(Test test);
-
+        Task<bool> IsTestNameNotUniqueAsync(string testName);    
```

Comment:
According to me, I think IsTestNameUnique, would be the better name, then to add 'Not' on the name. If we read the code, IsTestNameNotUnique, might confuse the reader. Let me know your thought on it.